### ISSUE TITLE
Campaign stops sending with status "Failed", but we can't figure out why

### ISSUE THREAD

**Issue by christofersandin** [putyourlightson/craft-campaign#286]
We have a setup where the campaigns are sent via Postmark. This works well in general but we have seen a strange issue with a few addresses where the send out fails instead of marking an address as bounced.

This happened 4 times with a list of approx 400 people today. We also had a few addresses that bounced and were properly marked up as such in Postmark and reported back to Campaign.

From the log:

```
…
2021-12-17 10:42:43 [][1][info] Sendout "Newsletter" initiated by "username".
2021-12-17 10:43:00 [][][info] Contact name1@company.com marked as "bounced".
2021-12-17 10:43:04 [][1][info] Sending of the sendout "Newsletter" to name2@company.com failed after 3 send attempt(s). Please check that your Campaign email settings are correctly configured and check the error in the Craft log.
2021-12-17 10:45:56 [][1][info] Sendout "Newsletter" initiated by "username".
2021-12-17 10:46:04 [][][info] Contact name3@company.com marked as "bounced".
2021-12-17 10:46:13 [][][info] Contact  name4@company.commarked as "bounced".
…
```

We can't see anything in the Craft log, and we can't see any mention of the affected addresses in the Postmark logs.

Any ideas where we can look for issues to resolve this? 

The only way to continue is to manually check the campaign.log file and then remove the address from the subscription list before hitting the "Prepare to Resume Sending" button in Craft again. This way of failing is more or less useless for our customer since we need to manually help them when this happens and the sendout got stuck on 30/400 sent emails.

Thanks!

**Comment by bencroker** [comment]
This likely has to do with contacts unsubscribing via Postmark directly, rather than through Campaign, meaning that sendouts will fail when sending to contacts that have already unsubscribed. This was resolved in https://github.com/putyourlightson/craft-campaign/issues/276, so please read the discussion there and let me know if this solves it for you.

**Comment by christofersandin** [comment]
Thanks, I can see that at least 3 out of the 4 addresses have the "Unsubscribed" status in Postmark. Didn't know Postmark had its own unsubscribe settings.

But, if I understand you correctly, the "Subscription Change" setting in Postmark here

<img width="272" alt="settings" src="https://user-images.githubusercontent.com/6552982/146555206-c4337153-c371-4bb8-9037-643ea9303351.png">

is not handled by default in Campaign but can be handled by us if we like (according to your comments in #276)?

**Comment by bencroker** [comment]
This must be a new setting as it didn't exist before. I'll look into adding this in the next release, but you'll probably anyway want to remove Postmark's own unsubscribe link from the email, as there's no point having multiple unsubscribe links. Campaigns unsubscribe link should ideally be maintained, as it will then be able to show unsubscribe metrics in the reports.

**Comment by christofersandin** [comment]
Great! 

And I agree, I see if I can figure out how to remove the Postmark ones and just use the Campaign links instead.

Thanks for the swift reply!

**Comment by christofersandin** [comment]
For reference, Postmark can help you remove the automatically inserted "Unsubscribe" link inserted to all Broadcast emails. 

This is through a manual process where [you need to contact them and explain how you deal with users who unsubscribe](https://postmarkapp.com/support/article/1208-how-to-add-an-unsubscribe-link).

**Comment by bencroker** [comment]
I added a "Subscription Change" event to the Postmark webhook that listens for unsubscribe events in https://github.com/putyourlightson/craft-campaign/commit/80a28764b4bf201358129e2e7bf4913b6d12649f, for good measure.

**Comment by bencroker** [comment]
Released in version 1.22.3.
